### PR TITLE
Top-level Makefile fix and Eigen issue

### PR DESCRIPTION
**Pull request by haradh1**
Once we get it in master, I'll backport them to 0.4.1.

Jira:MADLIB-589

It seems the wild card in Makefile rule doesn't work with fixed target
in some environment.  Also inlucde some pgxn package improvement (it will
be rewritten in cmake in the future.)

Jira:MADLIB-604

Eigen has been downloaded from Bitbucket, but now it seems it's better
to have it included in our source tree, as Bitbucket's subversion access
is not officially supported and there were a couple of reports that
it's not working these days.

### CODE REVIEW

**Comment by fschopp** [comment]
Not sure if adding binary files to the git repository is a good idea.

**Comment by haradh1** [comment]
Yeah, maybe we can mirror dependencies in our madlib.net or somewhere.  I initially thought about mirroring it in github, but git protocol is not graceful for us and it still needs HTTPS.

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -36,27 +36,22 @@ if(NOT BOOST_TAR_SOURCE)
```
Comment (fschopp) [inline src/CMakeLists.txt:35]: I'll ignore changes in this file and instead use pull request #183, OK?

Comment (haradh1) [inline src/CMakeLists.txt:35]: Sure.

**Pull request closed without merging**